main/options.c

```diff
@@ -2363,6 +2365,18 @@ static void processMaxRecursionDepthOption (const char *const option, const char
 	Option.maxRecursionDepth = atol(parameter);
 }
 
+static void processPatternLengthLimit(const char *const option, const char *const parameter)
+{
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "A parameter is needed after \"%s\" option", option);
+
+	if (atol (parameter) < 0)
```

Comment:
> Atoi returns 0 when a unconvertable string is given.

strtol does also return 0 for an unconvertable string. At least if you pass `NULL` as second argument.

Using strtol is not so straightforward one has to do something like

```c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
  char *ptr;
  const char input[] = "123";
  long val = strtol(input, &ptr, 10);

  printf("val = %d, ptr = %d\n", val, ptr);

  if(!*ptr)
    return val;

  return 1;
}
```
but currently no callers of strtol in options.c do that.

I've switched to `strtol` now.